src/struct_mv/boxloop_dispatcher.h

```diff
@@ -0,0 +1,29 @@
+/******************************************************************************
+ * Copyright (c) 1998 Lawrence Livermore National Security, LLC and other
+ * HYPRE Project Developers. See the top-level COPYRIGHT file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ ******************************************************************************/
+
+#ifndef HYPRE_BOXLOOP_DISPATCHER_HEADER
+#define HYPRE_BOXLOOP_DISPATCHER_HEADER
+
+#define CONCAT2(x, y) x ## y
+#define XCONCAT2(x, y) CONCAT2(x, y)
+
+#define HYPRE_BOXLOOP(BEGIN, BEGINARGS, BODY, END, ENDARGS)                                  \
```

Comment:
This is still just a draft idea, right?  Should we remove it from the PR?

Reply:
Yes, we should. Thanks!